controller/commentsController.js

```diff
@@ -6,6 +6,34 @@ const mongoose = require("mongoose");
 const CustomError = require("../utils/customError");
 const responseHandler = require("../utils/responseHandler");
 
+const validate = (value) => {
+  if (!mongoose.Types.ObjectId.isValid(value)) {
+    new CustomError(422, "invalid ID");
+    return;
+  }
+};
+
+exports.upvoteComment = async (req, res) => {
+  try {
+    const { commentId } = req.params;
+    const { ownerId } = req.body;
+
+    await validate(commentId);
+    const comment = await Comments.findById({ _id: commentId });
+    comment.vote = comment.vote + 1;
+    const commentResponse = comment.upVotes.includes(ownerId)
+      ? "Your upvote has already been registered"
+      : comment.upVotes.unshift(ownerId);
+    return res.json({
+      message: "Comment upVoted Successfully!",
+      response: "Ok",
+      data: commentResponse,
+    });
+  } catch (err) {
+    CustomError(err, res);
```

Comment:
Please pass the CustomError class to the next function. Also look at the parameters of the class and use them correctly. The CustomError class doesn't take any `res` object.

```suggestion
    return next(new CustomError(500, "Something went wrong, please try again later", err));
```